File: algorithms/cpp/maximumPopulationYear/maximumPopulationYear.cpp
// Source: https://leetcode.com/problems/maximum-population-year/
// Author: Miao Zhang
// Date:   2021-06-14

class Solution {
public:
    int maximumPopulation(vector<vector<int>>& logs) {
        vector<int> year(2051);
        for (auto& log: logs) {
            for (int i = log[0]; i < log[1]; i++) {
                year[i]++;
            }
        }
        int res = -1;
        int maxval = 0;
        for (int i = 1950; i < 2051; i++) {
            if (year[i] > maxval) {
                maxval = year[i];
                res = i;
            }
        }
        return res;
    }
};

File: algorithms/python/maximumPopulationYear/maximumPopulationYear.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/maximum-population-year/
# Author: Miao Zhang
# Date:   2021-06-14

class Solution:
    def maximumPopulation(self, logs: List[List[int]]) -> int:
        year = [0 for _ in range(2051)]
        for log in logs:
            for i in range(log[0], log[1]):
                year[i] += 1
        res = -1
        maxval = 0
        for i in range(1950, 2051):
            if year[i] > maxval:
                maxval = year[i]
                res = i
        return res
